### ISSUE TITLE
gcc src/main.c

### ISSUE THREAD

**Issue by i032** [SpongeData-cz/nativeextractor#4]
Occures exception like below when execute `gcc src/main.c`

```
/usr/bin/ld: /tmp/ccAecOVs.o: in function `main':
main.c:(.text+0x67): undefined reference to `extractor_c_new'
/usr/bin/ld: main.c:(.text+0xa3): undefined reference to `stream_file_c_new'
/usr/bin/ld: main.c:(.text+0xf3): undefined reference to `print_pos'
collect2: error: ld returned 1 exit status
```
_main.c code copied from Readme.md ("Testing miners" section)_

**Comment by mafiosso** [comment]
There is a mistake:
`gcc main.c `pkg-config --cflags --libs lnativeextractor` -o netest.bin` should change to
`gcc main.c `pkg-config --cflags --libs nativeextractor` -o netest.bin`, it works after `make install` only - of course

**Comment by i032** [comment]
Unfortunately it gave another error such as below

```
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib/libnativeextractor.so: undefined reference to `g_setenv'
...
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib/libnativeextractor.so: undefined reference to `g_unichar_isspace'
collect2: error: ld returned 1 exit status
```
It worked after adding -shared to end

`gcc src/main.c pkg-config --cflags --libs nativeextractor -o netest.bin -shared`

**Comment by mafiosso** [comment]
Shared option is not the right solution. You are missing some glib dependencies. You should link it by this `pkg-config glib-2.0 --cflags --libs`.

```bash
gcc src/main.c `pkg-config --cflags --libs nativeextractor glib-2.0` -o netest.bin
```

**Comment by michaelspongedata** [comment]
Fixed in commit 47c7310.
